README.md

```diff
@@ -223,3 +226,33 @@ function isMetaEqual(a, b) {
 }
 ```
 
+## Patch Application
+
+When you want to apply changes from one tree to another easily, you can use the
+`FSTree.applyPatch` method. For example, given:
+
+```js
+var patch = oldInputTree.calculatePatch(newInputTree);
+var inputDir = 'src';
+var outputDir = 'dist';
+FSTree.applyPatch(inputDir, outputDir, patch);
+```
+
+It will apply the patch changes to `dist` while using `src` as a reference for
+non-destructive operations (`mkdir`, `create`, `change`). You can optionally
+provide a delegate object to handle applying specific types of operations:
+
```

Comment:
Do we expect users to do anything with `patch` other than feed it back to `FSTree.applyPatch`?

I still like this API b/c it provides primitives that users can reason about, but it might make sense to have a helper `calculateAndApplyPatch(oldTree, srcDir, dstDir, delegate)`.  Not something to hold up this PR over though.